packages/client/src/router/resolveRoute.ts

```diff
@@ -0,0 +1,25 @@
+import { resolveRoutePath } from './resolveRoutePath.js'
+import type { PageMetaDefault, Route } from './routes.js'
+import { routes } from './routes.js'
+
+interface ResolvedRoute<PageMeta extends PageMetaDefault = PageMetaDefault>
+  extends Route<PageMeta> {
+  path: string
+}
+
+/**
+ * Resolve route with given path
+ */
+export const resolveRoute = <
+  PageMeta extends PageMetaDefault = PageMetaDefault,
+>(
+  path: string,
+): ResolvedRoute<PageMeta> => {
+  const routePath = resolveRoutePath(path)
+  const pageMapItem = routes.value[routePath] || routes.value['/404.html']
+
+  return {
+    ...(pageMapItem as ResolvedRoute<PageMeta>),
+    path: routePath,
+  }
+}
```

Comment:
This method is problematic, in a lot of cases, people expect to get `null` when the given path does not belong to any route.

I would suggest a second arguement being added:

```suggestion
export const resolveRoute = <
  PageMeta extends PageMetaDefault = PageMetaDefault,
>(
  path: string,
  fallback = false,
): ResolvedRoute<PageMeta> => {
  const routePath = resolveRoutePath(path)
  const pageMapItem = routes.value[routePath]

  if (!pageMapItem && !fallback) return {
    path: routePath,
    meta: null,
    loader: null,
  };

  return {
    ...(pageMapItem || routes.value['/404.html'] as ResolvedRoute<PageMeta>),
    path: routePath,
  }
}
```

Reply:
It's weird to have null meta and loader in a resolved route. I kind of prefer to return `ResolvedRoute<PageMeta>  | null`